addons/sale/static/src/js/sale_product_field.js

```diff
@@ -278,13 +284,42 @@ export class SaleOrderLineProductField extends ProductLabelSectionAndNoteField {
                 saleOrderRecord.data.order_line.leaveEditMode();
             },
             discard: () => {
-                saleOrderRecord.data.order_line.delete(this.props.record);
+                if (!selectedComboItems.length) {
+                    // Don't delete the main product if it's a combo product as it has been added
+                    // from combo configurator
+                    saleOrderRecord.data.order_line.delete(this.props.record);
+                }
             },
             ...this._getAdditionalDialogProps(),
         });
     }
 
-    async _openComboConfigurator(edit = false) {
+    async handleComboSave(comboProductData, selectedComboItems, edit, hasOptionalProducts) {
+        const saleOrder = this.props.record.model.root.data;
+        const comboLineRecord = this.props.record;
+        saleOrder.order_line.leaveEditMode();
+        const comboLineValues = {
+            product_uom_qty: comboProductData.quantity,
+            selected_combo_items: JSON.stringify(
+                selectedComboItems.map(serializeComboItem)
+            ),
+        };
+        if (!edit) {
+            comboLineValues.virtual_id = uuid();
+        }
+        await comboLineRecord.update(comboLineValues);
+        // Ensure that the order lines are sorted according to their sequence.
+        await saleOrder.order_line._sort();
+
+        if (hasOptionalProducts && !edit) {
+            const selectedComboProducts = selectedComboItems.map(
+                item => ({ name: item.product.display_name })
+            );
+            await this._openProductConfigurator(false, selectedComboProducts);
+        }
+    }
+
+    async _openComboConfigurator(edit = false, hasOptionalProducts = false) {
         const saleOrder = this.props.record.model.root.data;
         const comboLineRecord = this.props.record;
         const comboItemLineRecords = getLinkedSaleOrderLines(comboLineRecord);
```

Comment:
Don't forget to update this ;)

Reply:
I think you can simply do the following:
```suggestion
        const comboItemLineRecords = getLinkedSaleOrderLines(comboLineRecord).filter(line => !!line.data.combo_item_id);
```